src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,80 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { CommentData } from '../types/Comment';
+import { Post } from '../types/Post';
+
+type Props = {
+  addComment: (newComment: CommentData) => void;
+  buttonAddLoading: boolean;
+  selectedPost: Post | null;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  addComment,
+  buttonAddLoading,
+  selectedPost,
+}) => {
+  const [name, setName] = useState('');
+  const [nameError, setNameError] = useState(false);
+  const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState(false);
+  const [bodyComment, setBodyComment] = useState('');
+  const [bodyCommentError, setBodyCommentError] = useState(false);
+
+  let newComment: CommentData;
+
+  const handleFormSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    setNameError(false);
+    setEmailError(false);
+    setBodyCommentError(false);
+
+    let hasError = false;
+
+    if (name.trim().length === 0) {
+      setNameError(true);
+      hasError = true;
+    }
+
+    if (email.trim().length === 0) {
+      setEmailError(true);
+      hasError = true;
+    }
+
+    if (bodyComment.trim().length === 0) {
+      setBodyCommentError(true);
+      hasError = true;
+    }
+
+    if (hasError) {
+      return;
+    }
+
+    if (selectedPost) {
+      newComment = {
+        postId: selectedPost.id,
+        name: name,
+        email: email,
+        body: bodyComment,
+      };
+
+      addComment(newComment);
+      setBodyComment('');
+    }
+  };
+
+  const handleFormClear = () => {
+    setName('');
+    setEmail('');
+    setBodyComment('');
+    setNameError(false);
+    setEmailError(false);
+    setBodyCommentError(false);
+  };
```

Comment:
The function `handleFormClear` is resetting the form fields and their corresponding error states. This is a good practice. However, you could improve it by resetting the form using its `reset` method. This would automatically set all form fields to their initial values.